### PR TITLE
nautilus: mgr/dashboard: test prometheus rules through promtool

### PR DESCRIPTION
**Pull request by aaSharma14**
backport tracker: https://tracker.ceph.com/issues/49656

---

backport of

* https://github.com/ceph/ceph/pull/39246
* https://github.com/ceph/ceph/pull/40205 (one commit)
* https://github.com/ceph/ceph/pull/40300
* https://github.com/ceph/ceph/pull/40395

parent tracker: https://tracker.ceph.com/issues/45415

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Review by callithea — CHANGES_REQUESTED** [review]
Could you please remove the additional whitespace changes in `src/test/CMakeLists.txt`?

**Comment by aaSharma14** [comment]
> Could you please remove the additional whitespace changes in `src/test/CMakeLists.txt`?

Thanks @callithea . Done

**Comment by tchaikov** [comment]
is this a bug fix? if not, why do we need to backport it?

**Comment by aaSharma14** [comment]
> is this a bug fix? if not, why do we need to backport it?

This PR is adding unit tests for the prometheus rules. Tiil now we needed to manually do the testing for the alerts. Backporting this would be good IMO.

**Review by callithea — APPROVED** [review]

**Review by avanthakkar — APPROVED** [review]

**Review by tchaikov — CHANGES_REQUESTED** [review]
this change has been reworked on master. could please cherry-pick the follow-up changes as well?

**Comment by aaSharma14** [comment]
> this change has been reworked on master. could please cherry-pick the follow-up changes as well?

Thanks @tchaikov . Cherry-picked those changes as well. Please take a look.

File: install-deps.sh [hunk 1]
```diff
@@ -447,6 +450,35 @@ function activate_virtualenv() {
     . $env_dir/bin/activate
 }
 
+<<<<<<< HEAD
+=======
```
Comment (tchaikov) [inline install-deps.sh]: this change does not look right.

**Comment by tchaikov** [comment]
i don't think we need to install libboost 1.75 for building nautilus. neither do we need to build with libzbd on nautilus. so i'd suggest remove those commits from this backport PR.

**Comment by aaSharma14** [comment]
> i don't think we need to install libboost 1.75 for building nautilus. neither do we need to build with libzbd on nautilus. so i'd suggest remove those commits from this backport PR.

Thanks @tchaikov . I have removed those commits. Are the current commits okay for this backport? Or I need to change something else?

**Review by idryomov — CHANGES_REQUESTED** [review]
Please get rid of those merge commits by rebasing on latest nautilus and drop boost-related changes.  They might be needed but shouldn't be included in the dashboard backport PR.

**Review by idryomov — CHANGES_REQUESTED** [review]
You are missing "cherry-picked from" annotations, please use `git cherry-pick -x`.

Conflicts should be documented in the commit message -- what led to the conflict (i.e. what changes are missing in nautilus), how the conflict was resolved, etc.

**Comment by aaSharma14** [comment]
jenkins test make check

**Comment by aaSharma14** [comment]
jenkins test api

**Comment by aaSharma14** [comment]
> You are missing "cherry-picked from" annotations, please use `git cherry-pick -x`.
> 
> Conflicts should be documented in the commit message -- what led to the conflict (i.e. what changes are missing in nautilus), how the conflict was resolved, etc.

Thanks @idryomov . Will modify the commit message

**Review by idryomov — CHANGES_REQUESTED** [review]
`cherry picked from` annotations are still missing which suggests that you didn't use `git cherry-pick -x`.  The conflict resolution is messed up as well: your first commit is adding `<<<<<<< HEAD`, etc markers and then the second commit drops them.  These markers should not be there.

I suggest that you start from a clean slate, use `git cherry-pick -x` and make sure conflicts are resolved and documented properly in *each* commit.  Refer to https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst#cherry-picking-rules for details.

**Comment by idryomov** [comment]
@smithfarm Can you or someone else on the backports teams help Aashish with the process?

**Comment by smithfarm** [comment]
The first thing to clarify is: which commits from master need to be cherry-picked?

I see one commit from #39246 and then there is https://github.com/ceph/ceph/pull/40205 which seems to be related...

**Comment by smithfarm** [comment]
ah, I see the Pacific backport has 4 commits: #40929

maybe those 4, then?

**Comment by aaSharma14** [comment]
> `cherry picked from` annotations are still missing which suggests that you didn't use `git cherry-pick -x`. The conflict resolution is messed up as well: your first commit is adding `<<<<<<< HEAD`, etc markers and then the second commit drops them. These markers should not be there.
> 
> I suggest that you start from a clean slate, use `git cherry-pick -x` and make sure conflicts are resolved and documented properly in _each_ commit. Refer to https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst#cherry-picking-rules for details.

Thanks @idryomov Addressed your comments.

**Review by idryomov — COMMENTED** [review]
My concerns have been addressed.  Deferring to the Dashboard team -- the PR has changed quite a lot, please re-review.

**Review by nizamial09 — APPROVED** [review]

**Review by alfonsomthd — APPROVED** [review]

**Pull request merged**